backup-s3/src/main/scala/io/aiven/guardian/kafka/backup/s3/BackupClient.scala

```diff
@@ -200,30 +199,36 @@ class BackupClient[T <: KafkaConsumerInterface](maybeS3Settings: Option[S3Settin
       }
 
   private[s3] def kafkaBatchSink
-      : Sink[(UploadPartResponse, immutable.Iterable[kafkaClientInterface.CursorContext]), NotUsed] =
+      : Sink[(UploadPartResponse, immutable.Iterable[kafkaClientInterface.CursorContext]), Future[Done]] =
     // See https://pekko.apache.org/docs/pekko/current/stream/operators/Partition.html for an explanation on Partition
-    Sink.fromGraph(GraphDSL.create() { implicit builder =>
-      import GraphDSL.Implicits._
-      val partition = builder.add(
-        new Partition[(UploadPartResponse, immutable.Iterable[kafkaClientInterface.CursorContext])](
-          outputPorts = 2,
-          {
-            case (_: SuccessfulUploadPart, _) => 0
-            case (_: FailedUploadPart, _)     => 1
-          },
-          eagerCancel = true
-        )
-      )
-      partition.out(0) ~> successSink
-        .contramap[(UploadPartResponse, immutable.Iterable[kafkaClientInterface.CursorContext])] {
+    Sink.fromGraph(
+      GraphDSL.createGraph(
+        successSink.contramap[(UploadPartResponse, immutable.Iterable[kafkaClientInterface.CursorContext])] {
```

Comment:
This is the major difference which is that we pass in both the `successSink` and `failureSink` explicitly, this gives us access to the materialized values explained below.